src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -10235,11 +10235,15 @@ HCIMPL2(EXTERN_C void, ProfileEnter, UINT_PTR clientData, void * platformSpecifi
 
 #ifdef PROFILING_SUPPORTED
 
+    // We might have already torn down the profiler, if so we need to abort.
+    if (g_profControlBlock.curProfStatus.Get() != kProfStatusActive 
```

Comment:
What happens if the profiler is torn down right after we do this check, but before we actually call it below?

Reply:
Yeah, there are all sorts of races here. I realize now that even the check can race, depending on the order of events `g_profControlBlock.pProfInterface` could be set to null after `g_profControlBlock.pProfInterface.Load() == NULL` and then `g_profControlBlock.pProfInterface->GetEnterHook()          == NULL` would AV. I need to rework this entirely.